src/main/java/me/makkuusen/timing/system/heat/Heat.java

```diff
@@ -219,6 +220,10 @@ public boolean finishHeat() {
         if (getHeatState() != HeatState.RACING) {
             return false;
         }
+
+        HeatFinishEvent finishEvent = new HeatFinishEvent(this);
+        Bukkit.getServer().getPluginManager().callEvent(finishEvent);
```

Comment:
Any specific reason it is being called before all final changes has been completed? If it is not a problem it should probably be called after all manipulation of the heat object has been completed.

Reply:
You can move it to wherever you prefer, I just need to know when the heat has finished